src/main/java/org/kohsuke/github/GitHubClient.java

```diff
@@ -210,21 +210,31 @@ public boolean isAnonymous() {
     /**
      * Gets the current rate limit from the server.
      *
+     * For some versions of GitHub Enterprise, the {@code /rate_limit} endpoint returns a {@code 404 Not Found}. In
+     * that, if {@link #lastRateLimit()} is not {@code null} and is not expired, it will be returned. Otherwise, a
+     * placeholder {@link GHRateLimit} instance with {@link GHRateLimit.UnknownLimitRecord}s will be returned.
+     *
      * @return the rate limit
      * @throws IOException
      *             the io exception
      */
+    @Nonnull
     public GHRateLimit getRateLimit() throws IOException {
-        GHRateLimit rateLimit;
+        GHRateLimit result;
         try {
-            rateLimit = fetch(JsonRateLimit.class, "/rate_limit").resources;
+            result = fetch(JsonRateLimit.class, "/rate_limit").resources;
         } catch (FileNotFoundException e) {
-            // GitHub Enterprise doesn't have the rate limit
-            // return a default rate limit that
-            rateLimit = GHRateLimit.Unknown();
+            // GitHub Enterprise doesn't have the rate_limit endpoint
```

Comment:
I don’t think this statement is right anymore at least on recent versions from reading the docs,

My understanding is that it is available but there’s a global flag for turning it off or on.

Having it off means that it will return a 404